### ISSUE TITLE
Can not open file with file dialog using xdg-desktop-portal-kde

### ISSUE THREAD

**Issue by nolyn** [flatpak/flatpak#1957]
## Linux distribution and version
Arch linux

## Flatpak version
```
pacman -Qi flatpak
Name            : flatpak
Version         : 0.99.3-1
Description     : Linux application sandboxing and distribution framework
                  (formerly xdg-app)
Architecture    : x86_64
URL             : http://flatpak.org
Licenses        : LGPL2.1
Groups          : None
Provides        : None
Depends On      : dbus  systemd  glib2  libsoup  polkit  libxau  ostree
                  json-glib  libseccomp  libarchive  python  bubblewrap
                  appstream-glib  xdg-desktop-portal
Optional Deps   : None
Required By     : flatpak-builder
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 4.48 MiB
Packager        : Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
Build Date      : Wed 11 Jul 2018 01:02:44 AM CEST
Install Date    : Fri 10 Aug 2018 03:01:16 PM CEST
Install Reason  : Explicitly installed
Install Script  : Yes
Validated By    : Signature
```

## Description of the problem
I'm building a flatpak for a project in which i'm participating: https://github.com/IENT/YUView.
I managed to build the program surprisingly easy, but now I'm stuck on opening file when I run it. It is based on Qt. I have xdg-desktop-portal and  xdg-desktop-portal-kde and xdg-desktop-portal-gtk installed. I used the kde platform/sdk to do the build. 
The software is a viewer for raw video data files (among others). It is possible to add a file name as an argument to the call and this works. Drag and drop to the playlist also works.
However opening a file using the open file dialog does not work. I can browse and see the files, but when I open one nothing happens. I believe using the desktop portal does not require and changes to the code, it should just work out of the box for qt applications. Correct?
I tried to debug this problem following this: https://community.kde.org/Guidelines_and_HOWTOs/Flatpak

There is no output of xdg-desktop-portal-kde
```
QT_LOGGING_RULES='xdg-desktop*.debug=true' /usr/lib/xdg-desktop-portal-kde
xdp-kde: Desktop portal registered successfuly
```

The output of xdg-desktop-portal is:
```
 G_MESSAGES_DEBUG=all /usr/lib/xdg-desktop-portal                                                                                                                                              :(
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.814: loading /usr/share/xdg-desktop-portal/portals/kde.portal
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.814: loading /usr/share/xdg-desktop-portal/portals/gnome-shell.portal
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.814: loading /usr/share/xdg-desktop-portal/portals/gtk.portal
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.821: providing portal org.freedesktop.portal.NetworkMonitor
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.827: providing portal org.freedesktop.portal.ProxyResolver
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.827: Using gtk.portal for org.freedesktop.impl.portal.FileChooser in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.828: providing portal org.freedesktop.portal.FileChooser
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.828: Using gtk.portal for org.freedesktop.impl.portal.AppChooser in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.830: providing portal org.freedesktop.portal.OpenURI
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.830: Using gtk.portal for org.freedesktop.impl.portal.Print in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.831: providing portal org.freedesktop.portal.Print
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.831: Using gtk.portal for org.freedesktop.impl.portal.Screenshot in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.833: providing portal org.freedesktop.portal.Screenshot
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.833: Using gtk.portal for org.freedesktop.impl.portal.Notification in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.834: providing portal org.freedesktop.portal.Notification
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.834: Using gtk.portal for org.freedesktop.impl.portal.Inhibit in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.836: providing portal org.freedesktop.portal.Inhibit
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.836: Using gnome-shell.portal for org.freedesktop.impl.portal.Access in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.837: providing portal org.freedesktop.portal.Device
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.837: Using gtk.portal for org.freedesktop.impl.portal.Account in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.838: providing portal org.freedesktop.portal.Account
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.839: Using gtk.portal for org.freedesktop.impl.portal.Email in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.840: providing portal org.freedesktop.portal.Email
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.840: Using gtk.portal for org.freedesktop.impl.portal.ScreenCast in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.841: providing portal org.freedesktop.portal.ScreenCast
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.841: Using gtk.portal for org.freedesktop.impl.portal.RemoteDesktop in GNOME
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.842: providing portal org.freedesktop.portal.RemoteDesktop
** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:58:44.843: org.freedesktop.portal.Desktop acquired




** (/usr/lib/xdg-desktop-portal:25717): DEBUG: 20:59:06.295: convert uri file:///home/me/Videos/RaceHorsesL_832x480_30fps_8bit_420pf.yuv -> file:///run/user/1000/doc/5463d076/RaceHorsesL_832x480_30fps_8bit_420pf.yuv
```

The output of xdg-desktop-portal-gtk is:
```
G_MESSAGES_DEBUG=all /usr/lib/xdg-desktop-portal-gtk -v    
XDP: providing org.freedesktop.impl.portal.FileChooser
XDP: providing org.freedesktop.impl.portal.AppChooser
XDP: providing org.freedesktop.impl.portal.Print
XDP: providing org.freedesktop.impl.portal.Screenshot
XDP: providing org.freedesktop.impl.portal.Notification
XDP: Using org.gnome.SessionManager for inhibit
XDP: providing org.freedesktop.impl.portal.Inhibit
XDP: providing org.freedesktop.impl.portal.Access
XDP: providing org.freedesktop.impl.portal.Account
XDP: providing org.freedesktop.impl.portal.Email
XDP: org.freedesktop.impl.portal.desktop.gtk acquired

** (xdg-desktop-portal-gtk:25677): WARNING **: 20:59:04.547: Failed to create foreign window for XID 1729136793

** (xdg-desktop-portal-gtk:25677): WARNING **: 20:59:04.547: Failed to associate portal window with parent window x11:67108899

** (xdg-desktop-portal-gtk:25677): WARNING **: 21:21:03.708: Failed to create foreign window for XID 1729136793

** (xdg-desktop-portal-gtk:25677): WARNING **: 21:21:03.708: Failed to associate portal window with parent window x11:67108899
```
These warings appear after opening the open file dialog. Is the gtk portal used instead of the kde portal? I'm starting it from a gnome session. I used kde sdk to built it since I read it includes Qt and the project is qt based. 
Maybe this is related: #1307 

Please let me know if I can provide further information. Hope someone can help me figure this out :)

**Comment by alexlarsson** [comment]
@grulja Any idea?

Also, exactly which kde runtime version are you using?

**Comment by grulja** [comment]
It will use xdg-desktop-portal-gtk (Gnome) portal backend since you use it under Gnome. If you would like to use the KDE one, you would need to set "XDG_CURRENT_DESKTOP=KDE" when starting xdg-desktop-portal to make it to look for KDE backend implementation. On the other hand, it doesn't make sense to use it, you want native file dialog, right? So you should be fine with the gtk one. Regarding your problem other problem, can you run "dbus-monitor --session" and keep it opened and try to open a file? Then please attach the output here.

**Comment by nolyn** [comment]
so I don't even need the kde portal, good to know that. Here is the dbus log:

[dbus-monitor.log](https://github.com/flatpak/flatpak/files/2286994/dbus-monitor.log)

My flatpak config looks like this:
```
app-id: de.ient.YUView
runtime: org.kde.Platform
runtime-version: 5.11
sdk: org.kde.Sdk
command: YUView
finish-args:
  - --socket=wayland
  - --socket=x11
  - --share=ipc
  - --device=dri
  - --filesystem=host 
  - --share=network 
  - --env=PATH=/usr/bin:/app/bin:/app/bin/x86_64-linux
build-options:
  env:
    LD_LIBRARY_PATH: /app/lib:/app/lib64
...
```
I didn't push everything related to github yet, since it is not working yet, but I could do that if it helps. 
I just tried another flatpak based on the kde platform: org.kde.okular and I'm able to open pdf files with the file dialog. 
I tried both system-wide and local install (--user), makes no difference.
I just tried two other features, YUView can save a playlist (does not work) and screen shotshots (does work!?). I'll compare the code and see if I can work out what the difference is. I already checked it quickly and as far as I could see it just uses plain QFileDialog.

**Comment by nolyn** [comment]
I can saving playlists to work if I already add the correct file ending when specifying the file. If it is not there it is added automatically. I guess this modification of the path is not allowed by the portals?

**Comment by grulja** [comment]
Can I try your application somehow? Can you share complete manifest?

**Comment by nolyn** [comment]
While the two save functions use the static QFileDialog::getSaveFileName, the open file code creates a QFileDialog object first.

**Comment by nolyn** [comment]
yes, I can push the related stuff to github and then upload the complete manifest here

**Comment by nolyn** [comment]
Here is the manifest: [de.ient.YUView.yaml.txt](https://github.com/flatpak/flatpak/files/2287368/de.ient.YUView.yaml.txt)

I tested the manifest again after changing it to use the sources from github, now I can suddenly open files! I don't understand it. Even if something was wrong with the cached files and I also used --force-clean --disable-cache a lot the config doesn't mention anything about the portals. And I had the portals installed from the beginning ...

Anyway, I'll now look into getting it to run on our workplace system, different from my laptop and if it still works this can be closed.
